File: TARS/models/generic_llms/openai.py
from langchain_openai import ChatOpenAI
from TARS.config.config import openai_settings
from dotenv import load_dotenv
from pydantic import SecretStr

load_dotenv()

openai_llm = ChatOpenAI(model=openai_settings.openai_model, temperature=openai_settings.openai_temperature, api_key=SecretStr(openai_settings.openai_api_key))

File: TARS/models/generic_llms/openai.js
const { Configuration, OpenAIApi } = require('openai');
const dotenv = require('dotenv');

dotenv.config();

const configuration = new Configuration({
  apiKey: process.env.OPENAI_API_KEY,
});

const openai = new OpenAIApi(configuration);

const openaiLlm = {
  async generateResponse(prompt) {
    try {
      const response = await openai.createCompletion({
        model: process.env.OPENAI_MODEL,
        prompt: prompt,
        max_tokens: 150,
        temperature: 0.7,
      });
      return response.data.choices[0].text.trim();
    } catch (error) {
      console.error('Error generating response:', error);
      return 'An error occurred while generating the response.';
    }
  },
};

module.exports = openaiLlm;
